Parsimony/ParserBuilder/PropertySelector.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace Parsimony.ParserBuilder
+{
+    internal class PropertySelector<TInput, TProp>
+    {
+        /// <summary>
+        /// The name of the member the selector function targets
+        /// </summary>
+        public string MemberName { get; }
+
+        /// <summary>
+        /// A setter for the property 
+        /// </summary>
+        public Action<TInput, TProp> Setter { get; }
+
+        /// <summary>
+        /// A getter for the property
+        /// </summary>
+        public Func<TInput, TProp> Getter { get; }
+
+        /// <summary>
+        /// Constructs a new PropertySelector with the given selector function
+        /// </summary>
+        /// <exception cref="ArgumentNullException"> <paramref name="selector"/> is null </exception>
+        /// <exception cref="ArgumentException"> <paramref name="selector"/> is not of type <see cref="MemberExpression"/> </exception>
+        /// <exception cref="ArgumentException"> The member selected by <paramref name="selector"/> is not a property </exception>
+        /// <exception cref="ArgumentException"> The member selected by <paramref name="selector"/> is not readable and writable </exception>
+        /// <param name="selector"> A property selector function</param>
+        public PropertySelector(Expression<Func<TInput, TProp>> selector)
+        {
+            if (selector == null)
+                throw new ArgumentNullException(nameof(selector));
+
+            if (!(selector.Body is MemberExpression mbr))
+                throw new ArgumentException("Must be a member selector expression");
+
+            //TODO:CN -- Handle convert nodes? casts etc?
+            //TODO:CN -- Handle public mutable fields?
```

Comment:
I think by the time we're using reflection we probably don't care about mutability, just access modifier. As a dev/client I'd rather have the framework be able to set "immutable" stuff so that the value it returns can immutable.

Reply:
I'm pretty sure even if I want to I can't reflect off a setter for that. Initially I just didn't want to implement for both public fields and properties because the scope is already too big

Reply:
Created issue #9 to address this